examples/datasets/generate_missing_values.py

```diff
@@ -0,0 +1,107 @@
+"""
+=============================================
+Generating MNAR / MCAR missing_values in data
+=============================================
+
+This example illustrates how the :func:`sklearn.datasets.drop_values` can
+be used to generate missing values that are correlated/non-correlated
+with the target.
+
+This function provisions generating missing values incrementally so that an
+exact fraction of missing values can be introduced for benchmarking
+missing-value handling strategies and evaluating the performance of such
+strategies with respect to the type and extent of missingness in data.
+
+MNAR or Missing Not At Random refers to the case when the missingness in the
+data is correlated with the target value(s).
+
+MCAR or Missing Completely At Random refers to the case when the missingness
+in the data is completely random and does not correlate with the target
+value(s).
+"""
+# Author: Raghav RV <rvraghav93@gmail.com>
+#
+# License: BSD 3 clause
+
+from __future__ import print_function
+from sklearn.datasets import drop_values
+
+print(__doc__)
+
+
+X = [[0, 1, 2],
+     [3, 4, 5],
+     [6, 7, 8],
+     [9, 0, 1],
+     [2, 3, 4],
+     [8, 9, 8],
+     [1, 0, 5],
+     [7, 8, 9],
+     [5, 4, 3],
+     [2, 1, 1],
+     [3, 4, 5],
+     [2, 3, 4],
+     [8, 9, 8],
+     [1, 0, 5],
+     [7, 8, 9]]
+y = [0, 1, 1, 1, 1, 0, 0, 0, 0, 1, 2, 2, 2, 2, 2]
+
+# Drop 10% of values across all features for samples
+# where the target class label is randomly chosen
+
+X, y, labels = drop_values(X, y,
+                           drop_fraction=0.1,
+                           correlated_labels=1,
+                           return_correlated_labels=True,
+                           copy=False,
+                           random_state=42)
+
+print("\nAfter dropping 10%% of values when class label(s) are %r\n" % labels)
+print("y \t X")
+print("------------------------")
+for i in range(y.shape[0]):
+    print(y[i], '\t', X[i])
```

Comment:
(Small thing: you can use `print(y[i], X[i], sep='\t')`. Then you don't also get a space between the y and the tab, and the tab and the X)